platform-api/src/main/java/io/datacater/core/deployment/DeploymentEndpoint.java

```diff
@@ -1,49 +1,162 @@
 package io.datacater.core.deployment;
 
-import com.fasterxml.jackson.core.JsonProcessingException;
+import io.datacater.core.pipeline.PipelineEntity;
+import io.datacater.core.stream.StreamEntity;
+import io.fabric8.kubernetes.api.model.ListMeta;
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.dsl.LogWatch;
+import io.fabric8.kubernetes.client.dsl.RollableScalableResource;
 import io.smallrye.mutiny.Uni;
-import java.util.List;
+import java.io.*;
+import java.util.UUID;
 import javax.annotation.security.RolesAllowed;
+import javax.enterprise.context.RequestScoped;
 import javax.inject.Inject;
-import javax.persistence.criteria.CriteriaBuilder;
-import javax.persistence.criteria.CriteriaDelete;
-import javax.persistence.criteria.Root;
 import javax.ws.rs.*;
+import javax.ws.rs.core.Context;
 import javax.ws.rs.core.MediaType;
-import org.eclipse.microprofile.openapi.annotations.parameters.RequestBody;
-import org.eclipse.microprofile.openapi.annotations.security.SecurityRequirement;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.sse.OutboundSseEvent;
+import javax.ws.rs.sse.Sse;
+import javax.ws.rs.sse.SseEventSink;
 import org.hibernate.reactive.mutiny.Mutiny;
+import org.jboss.logging.Logger;
 
 @Path("/api/alpha/deployments")
 @RolesAllowed("dev")
 @Produces(MediaType.APPLICATION_JSON)
-@SecurityRequirement(name = "apiToken")
+@RequestScoped
 public class DeploymentEndpoint {
+  private static final Logger LOGGER = Logger.getLogger(DeploymentEndpoint.class);
   @Inject Mutiny.SessionFactory sf;
 
+  @Inject KubernetesClient client;
+
+  @GET
+  @Path("{deploymentName}")
+  public Uni<ObjectMeta> getStream(@PathParam("deploymentName") String deploymentName) {
+    return Uni.createFrom().item(getK8Deployment(deploymentName));
+  }
+
+  @GET
+  @Path("{deploymentName}/logs")
+  @Produces(MediaType.TEXT_PLAIN)
+  public Uni<String> getLogs(@PathParam("deploymentName") String deploymentName) {
+    return Uni.createFrom().item(getDeploymentLogs(deploymentName));
+  }
+
+  @GET
+  @Path("{deploymentName}/watchLogs")
```

Comment:
In the path, please use lower-cased words with a hyphen as separator. Here, it would boil down to `{deploymentName}/watch-logs`.